File: tests/machine/x/go/user_type_literal.go
//go:build ignore

package main

import (
	"fmt"
)

type Person struct {
	Name string `json:"name"`
	Age  int    `json:"age"`
}

type Book struct {
	Title  string `json:"title"`
	Author Person `json:"author"`
}

func main() {
	var book Book = Book{"Go", Person{"Bob", 42}}
	_ = book
	fmt.Println(book.Author.Name)
}

File: tests/transpiler/x/php/user_type_literal.php
<?php
$book = ["title" => "Go", "author" => ["name" => "Bob", "age" => 42]];
echo rtrim((is_float($book["author"]["name"]) ? sprintf("%.15f", $book["author"]["name"]) : $book["author"]["name"])), PHP_EOL;
?>
